src/Main.elm

```diff
@@ -52,51 +52,129 @@ type DecimalStatus
 executeAdd : Model -> Model
 executeAdd model =
     { model
-        | display = String.fromFloat <| model.valueA + model.valueB
+        | display = showResults <| model.valueA + model.valueB
         , valueA = 0
-        , actionA = Blank
+        , actionA = Nothing
         , valueB = 0
-        , actionB = Blank
+        , actionB = Nothing
         , displayStatus = ShowingResults
     }
 
 
+executeSubtract : Model -> Model
+executeSubtract model =
+    { model
+        | display = String.fromFloat <| model.valueA - model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+executeMultiply : Model -> Model
+executeMultiply model =
+    { model
+        | display = String.fromFloat <| model.valueA * model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+executeDivide : Model -> Model
+executeDivide model =
+    { model
+        | display = String.fromFloat <| model.valueA / model.valueB
+        , valueA = 0
+        , actionA = Nothing
+        , valueB = 0
+        , actionB = Nothing
+        , displayStatus = ShowingResults
+    }
+
+
+showResults : Float -> String
+showResults float =
+    let
+        string =
+            String.fromFloat float
+
+        hasDecimal =
+            String.any (\a -> '.' == a) string
+
+        largerThenTen =
+            case hasDecimal of
+                True ->
+                    String.length string > 11
+
+                False ->
+                    String.length string > 10
+    in
+    case ( string, largerThenTen ) of
+        ( text, False ) ->
+            text
+
+        ( _, True ) ->
+            "ERROR"
+
+
 
 -- Update
 
 
 type Msg
     = UpdateDisplay CalButton
     | Saveinput CalButton
+    | AllClear
 
 
-buttonToAction : CalButton -> Action
+buttonToAction : CalButton -> Maybe Action
```

Comment:
This function feels very weird now. Probably there is someplace where you are pattern matching on `CalButton` already and could capture the `Action` that is contained inside to pass down to wherever this function is being used.

Reply:
Now using `Action` type directly and not using `Calbutton` to get to 'Action'